test_runner/regress/test_pageserver_layer_rolling.py

```diff
@@ -37,12 +40,20 @@ async def run_worker_for_tenant(
         finally:
             await conn.close(timeout=10)
 
-        last_flush_lsn = Lsn(ep.safe_psql("SELECT pg_current_wal_flush_lsn()")[0][0])
+        loop = asyncio.get_running_loop()
+        sql = await loop.run_in_executor(
+            None, lambda ep=ep: ep.safe_psql("SELECT pg_current_wal_flush_lsn()")
+        )
+        last_flush_lsn = Lsn(sql[0][0])
         return last_flush_lsn
 
 
 async def run_worker(env: NeonEnv, tenant_conf, entries: int) -> tuple[TenantId, TimelineId, Lsn]:
-    tenant, timeline = env.create_tenant(conf=tenant_conf)
+    loop = asyncio.get_running_loop()
+    # capture tenant_conf by specifying `tenant_conf=tenant_conf`, otherwise it will be evaluated to some random value
+    tenant, timeline = await loop.run_in_executor(
+        None, lambda tenant_conf=tenant_conf, env=env: env.create_tenant(conf=tenant_conf)
```

Comment:
The neon_local parts aren't safe to call concurrently (here in `create_tenant` and also where `run_worker_for_tenant` uses `env.endpoints.create_start`.  Also, that create_start() call probably needs wrapping in run_in_executor too?

I played a little with this code, and concluded that it's probably simpler to make neon_local enforce its own safety rather than do a mutex here: https://github.com/neondatabase/neon/pull/10185